src/output-json-alert.c

```diff
@@ -791,15 +791,37 @@ static int AlertJsonDecoderEvent(ThreadVars *tv, JsonAlertLogThread *aft, const
             continue;
         }
 
-        JsonBuilder *jb = jb_new_object();
-        if (unlikely(jb == NULL)) {
+        JsonBuilder *jb =
+                CreateEveHeader(p, LOG_DIR_PACKET, "alert", NULL, json_output_ctx->eve_ctx);
+        if (unlikely(jb == NULL))
             return TM_ECODE_OK;
+
+        AlertJsonHeader(p, pa, jb, json_output_ctx->flags, NULL, NULL);
+
+        if (PacketIsTunnel(p)) {
+            AlertJsonTunnel(p, jb);
         }
 
-        /* just the timestamp, no tuple */
-        jb_set_string(jb, "timestamp", timebuf);
+        /* payload */
+        if (json_output_ctx->flags &
```

Comment:
this seems  useless, as we won't have a payload for this case?

Reply:
Not even for the `decode-event:ipv6.frag_pkt_too_large` kind?

Reply:
I think that would put us in the first condition here?
```c
    if (PacketIsIPv4(p) || PacketIsIPv6(p)) { 
        return AlertJson(tv, aft, p);
    } else if (p->alerts.cnt > 0) { 
        return AlertJsonDecoderEvent(tv, aft, p);
    }
```